### PR TITLE
refactor: improve axios connection

### PR DESCRIPTION
**Pull request by wa0x6e**
- Improve the axios fetch request by adding timestamp and keepAlive
- Refactor image resolvers by extracting the axios image fetcher to utils.ts

### CODE REVIEW

File: src/resolvers/space.ts [hunk 1]
```diff
@@ -12,12 +13,13 @@ export default async function resolve(key) {
         method: 'post',
         data: {
           query: `query { space(id: "${key}") { avatar } }`
-        }
+        },
+        ...axiosDefaultParams
```
Comment (ChaituVR) [inline src/resolvers/space.ts:17]: I think we should pass `axiosDefaultParams` in `fetchHttpImage` itself, is there any reason to import it here on all resolvers? 🤔

File: src/resolvers/utils.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    await axios({
+      url,
+      ...{
+        responseType: 'arraybuffer',
+        timeout: 5e3,
+        ...options
+      }
+    })
+  ).data;
+}
```
Comment (ChaituVR) [inline src/resolvers/utils.ts:16]: I mean like this
```typescript
export async function fetchHttpImage(url: string, options = {}): Promise<Buffer> {
  return (
    await axios({
      url,
      ...{
        responseType: 'arraybuffer',
        timeout: 5e3,
        ...options,
        ...axiosDefaultParams
      }
    })
  ).data;
}
```

we can also make it optional based on some param passed to `fetchHttpImage`

**Review by ChaituVR — CHANGES_REQUESTED** [review]

File: src/resolvers/utils.ts (hunk 2 above)
Comment (ChaituVR) [reply src/resolvers/utils.ts:16]: Only problem is future generations may not remember about `axiosDefaultParams` 😋

Comment (wa0x6e) [reply src/resolvers/utils.ts:16]: `fetchImage` does not need keepalive, as I don’t think it has benefits when the url is not from same host. We could still uses it there though

File: src/resolvers/space.ts (hunk 1 above)
Comment (wa0x6e) [reply src/resolvers/space.ts:17]: These params are injected in the graphql axios query, and not fetchImate

Comment (ChaituVR) [reply src/resolvers/space.ts:17]: ahh makes sense. i thought its the same

File: src/resolvers/utils.ts [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+export const axiosDefaultParams = {
+  httpAgent: new http.Agent({ keepAlive: true }),
+  httpsAgent: new https.Agent({ keepAlive: true }),
+  timeout: 5e3
+};
+
+export async function fetchHttpImage(url: string): Promise<Buffer> {
+  return (
+    await axios({
```
Comment (ChaituVR) [inline src/resolvers/utils.ts:13]:
```suggestion
  return (
    axios({
```

**Review by ChaituVR — APPROVED** [review]
utAck

File: src/resolvers/utils.ts (hunk 3 above)
Comment (wa0x6e) [reply src/resolvers/utils.ts:13]: We need this one, else the `.data` will not resolve

**Pull request merged**